### PR TITLE
New image details page

### LINKED ISSUE #529: New single image result page
The single image result page has been redesigned. This page will be presented from a modal when navigated client-side, but that will be handled in a separate PR (#480).

## References

- **Figma of mobile:** https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Openverse-Releases?node-id=958%3A19663

- **Figma of desktop:** https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Openverse-Releases?node-id=1228%3A21557

## Related
 
- #480
- #496
- #511

### LINKED ISSUE #710: Horizontal scroll issues on single image result view
## Description

The current single image result view currently has a horizontal scrollbar: it's small on desktop and much more noticeable on large displays:

CleanShot 2022-01-25 at 08 37 06@2x

CleanShot 2022-01-25 at 08 37 16@2x

## Expectation

This will be fixed by #682, or we could push a more emergency fix if that needs time.

## Environment

 - Device: 
 - OS: 
 - Browser: 
 - Version: 
 - Other info: 

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### LINKED ISSUE #737: Image attributions load incorrectly on client-side routing
## Description

When navigating to a single image page directly, the attribution text loads correctly:

https://wordpress.org/openverse/image/3fa11800-4907-43d0-9921-52ed5ddb22aa

When visited client side from a search results page, the license name does not display! Click any image here and observe: https://wordpress.org/openverse/search/?q=dogs

CleanShot 2022-01-27 at 10 56 29@2x

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #529 by @zackkrida
Fixes #710 by @zackkrida
Fixes #737 by @zackkrida

## Description

This PR applies the new styles for the image results and rewrites some components to generalize to "media" or use one specific for images.

### Pending after this PR
- ~SketchFab viewer~
- Content report
- Show image tags? Yes
- Survey of usage?

## Testing Instructions

Run the project, go to the details page of any image and compare with mockups (Figma links in the issue associated). 

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable). **Still, there is work pending here.**
- [N/A] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/pages/image/_id.vue [hunk 1]
```diff
@@ -1,48 +1,125 @@
...
-  <div :aria-label="$t('photo-details.aria.main')">
-    <VPhotoDetails
+  <div>
+    <a
+      v-if="shouldShowBreadcrumb"
+      class="px-6 py-4 flex flex-row items-center bg-dark-charcoal-06 font-semibold text-xs md:text-sr"
+      :href="breadcrumbUrl"
+      @click.prevent="$router.back()"
+    >
+      <Chevron class="-ml-2" />
```
Comment (obulat) [inline src/pages/image/_id.vue:9]:
```suggestion
      <Chevron class="-ms-2" />
```

Comment (krysal) [reply src/pages/image/_id.vue:9]: Thanks!

**Comment by zackkrida** [comment]
cc @panchovm, I know the image corners are meant to be `rounded-sm`, but in the mockups it's currently all 4 image corners rounded. Would only the top corners be better?

Example: https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Openverse-Releases?node-id=1228%3A21557

Also need to think about the 'back to search results' link and how it should be styled, as that was not present in the mockups originally.

I can make a PR for you to reintroduce the Sketchfab viewer!

**Comment by panchovm** [comment]
I noticed the following on desktop:
1. Margin top of area where image title and CTA are is bigger than mockup. It should be `24px`.
2. Main button (CTA) text should be `Heading 5`style and paddings `16px 24px`.
2. Links should have an underline style on `hover`. 
3. License info should have `Description regular` style in license mention and CC icons description.
4. Spacing between license mention and CC icons description should be `16px`.
5. CC license icon should be `26px` square size. It seems that is not framed in a square view, as seen in [this screenshot](https://user-images.githubusercontent.com/895819/151030002-935a11f3-c231-42ad-838d-811eb4696b0f.png).
6. Report action should say `Report this content` instead of `Report`. I am not including the disabled state of the action as it seems a WIP.
7. In image information, data category is smaller than the data below. It should be the same, `Description regular` style.

And the following on mobile:
1. On a viewport smaller than `500px` the page starts masking content instead of adapting it. It looks [like this](https://cln.sh/ThIsrg).
2. Main button (CTA) text should be `Caption bold`style and top and bottom paddings `12px`.
2. Both image title and author should be `Description bold` style. They have the same font size but a different line-height.
3. **License** and **Credit the creator** texts should be `Description bold`.
4. Paddings of active tab in attribution box should be `12px 16px`.
5. Padding in attribution box should be `16px`.
5. Texts of **License** (description and CC icons definition), **Credit the creator**, and **Image information** should be either `Label regular` or `Label bold`. It's not consistent among the regular and bold versions.

Finally, some other notes.
* We need to define max values for width and height in the area where main image is placed. Otherwise, the image covers the whole screen. I suggest `1000px` for width and `500px` for height where the image scales up keeping its ratio until reaching one of the two values.
* Setting rounded corners only for top-left and top-right makes sense @zackkrida. Let's go with that. I will update the mockup 
* On mobile, I think we should keep the meta-data info in two columns as some categories can have long texts.

**Comment by zackkrida** [comment]
@krysal I'm noticing this PR has the same critical bug found in #737, this could potentially be a good place to fix it.
This looks beautiful!

**Comment by zackkrida** [comment]
@panchovm The new single image view does not show image _tags_, as present in the current site:

CleanShot 2022-01-27 at 14 14 05@2x

Do you think we should add these, similar to the _audio_ results page:

CleanShot 2022-01-27 at 14 14 41@2x

**Comment by krysal** [comment]
Right, I think I have addressed all your comments @panchovm and so the page is ready for another review. The problem with small viewports is due to the logo text appearing in the header which is being addressed in #718 I believe, if not it's better to solve that separately. I removed the header for you to take a look at how the page looks without the horizontal scroll, I'll add it again just before merging this.

@zackkrida The issue with the missing license should be solved here too.

File: src/components/VImageDetails/VImageDetails.vue [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    const { store } = useContext()
+
+    const imgType = computed(() => {
+      if (props.imageType) {
+        if (props.imageType.split('/').length > 1) {
+          return props.imageType.split('/')[1].toUpperCase()
+        }
+        return props.imageType
+      }
+      return 'Unknown'
```
Comment (sarayourfriend) [inline src/components/VImageDetails/VImageDetails.vue:83]: Does this need a translation or is it handled elsewhere?

File: src/components/VImageDetails/VRelatedImages.vue [hunk 3]
```diff
@@ -21,10 +19,10 @@
 import { ref } from '@nuxtjs/composition-api'
 import useRelated from '~/composables/use-related'
 import { IMAGE } from '~/constants/media'
-import ImageGrid from '~/components/ImageGrid/ImageGrid'
+import ImageGrid from '~/components/ImageGrid/ImageGrid.vue'
```
Comment (sarayourfriend) [inline src/components/VImageDetails/VRelatedImages.vue:22]: Should we switch this to use the new ImageGrid from the search results page?

File: src/components/VImageDetails/VRelatedImages.vue [hunk 4]
```diff
@@ -50,6 +41,13 @@ export default {
     if (props.service) {
       relatedOptions.service = props.service
     }
+    /**
+     * Fetches related images on `imageId` change
+     * @param {object} props
+     * @param {string} props.imageId
+     * @param {any} props.service
+     * @return {{ images: Ref<ImageDetail[]> }}
+     */
```
Comment (sarayourfriend) [inline src/components/VImageDetails/VRelatedImages.vue:50]: Just a heads up that this syntax will only work for a function definition, not a function call. Should we move this annotation to where `useRelated` is defined?

File: src/components/VMediaInfo/VCopyLicense.vue [hunk 5]
```diff
@@ -0,0 +1,233 @@
+<template>
+  <div>
+    <h5 class="mb-4 text-base md:text-2xl font-semibold">
+      {{ $t('media-details.reuse.copy-license.title') }}
+    </h5>
+
+    <nav class="flex borber-b-none" role="tablist">
+      <button
```
Comment (sarayourfriend) [inline src/components/VMediaInfo/VCopyLicense.vue:8]: Just a note but this pattern isn't accessible and doesn't follow the WAI-ARIA spec for accessible tabs. I've opened this issue to create a component for it: https://github.com/WordPress/openverse-frontend/issues/749

File: src/pages/image/_id.vue [hunk 6]
```diff
@@ -1,48 +1,127 @@
...
+import VIcon from '~/components/VIcon/VIcon.vue'
 
-const PhotoDetailPage = {
-  name: 'PhotoDetailPage',
-  components: { RelatedImages, VPhotoDetails },
+import Chevron from '~/assets/icons/chevron-left.svg?inline'
+
+const VImageDetailsPage = {
+  name: 'VImageDetailsPage',
+  layout: 'blank',
```
Comment (sarayourfriend) [inline src/pages/image/_id.vue:101]: I'Using the `blank` layout is causing RTL to break. Is this just temporary to remove the header for reviewing and we'll switch back to `default` after?

File: test/unit/specs/components/ImageDetails/v-copy-license.spec.js [hunk 7]
```diff
@@ -54,12 +54,12 @@ describe('CopyLicense', () => {
...
 
   it('should contain the correct contents', () => {
-    const wrapper = render(CopyLicense, options)
+    const wrapper = render(VCopyLicense, options)
     expect(wrapper.find('.copy-license')).toBeDefined()
   })
 
-  it('should dispatch COPY_ATTRIBUTION', () => {
-    const wrapper = render(CopyLicense, options)
+  it.skip('should dispatch COPY_ATTRIBUTION', () => {
```
Comment (sarayourfriend) [inline test/unit/specs/components/ImageDetails/v-copy-license.spec.js:61]: Will we be able to un-skip these tests? If not can we add a comment with a note for why they're skipped and an issue to fix the problem soon?

**Review by sarayourfriend — COMMENTED** [review]
Looking really great so far!

**Comment by panchovm** [comment]
Here my comments:

### On desktop
1. I am still seeing a bigger `margin-top` of the area where image title and CTA are. It should be `24px`.
2. License description looks good now, but each bullet of CC icon description is not `Description regular`.
3. Spacing between each CC icon description should be `16px`.
4. CC license icon should be `26px` square size.
5. **Copy text** button has incorrect padding, it should be `8px 12px`.

### On mobile
1. It seems that CC icon and license description alignment is not vertical. [Like this](https://cln.sh/GW1EyA).
2. Active tab padding in the attribution box seems irregular, top is `12px` and bottom `10px`. If it needs to be like that for a functional reason, that's fine. [Here is](https://cln.sh/TAjUYU) what I'm talking about.

The progress looks awesome. Great work ✨

File: src/components/VMediaInfo/VCopyLicense.vue (hunk 5 above)
Comment (krysal) [reply src/components/VMediaInfo/VCopyLicense.vue:8]: Thank you!

File: src/components/VImageDetails/VRelatedImages.vue (hunk 3 above)
Comment (krysal) [reply src/components/VImageDetails/VRelatedImages.vue:22]: It should definitely change once we have the new grid with masonry layout, but I don't see a new `VImageGrid` component yet. Do you mean the `VImageCell`? It's only for the square layout of all search results I believe.

File: src/components/VImageDetails/VImageDetails.vue (hunk 2 above)
Comment (krysal) [reply src/components/VImageDetails/VImageDetails.vue]: Needs translation, good catch! Fixed in [`7900ba7`](https://github.com/WordPress/openverse-frontend/pull/682/commits/7900ba79d81e406d2114fe4148e88e392d73f177).

File: src/pages/image/_id.vue (hunk 6 above)
Comment (krysal) [reply src/pages/image/_id.vue]: Yes, it was a temporary workaround to show the page without the horizontal scroll that the header was causing but it's already fixed so now it's back to the default layout.

File: test/unit/specs/components/ImageDetails/v-copy-license.spec.js (hunk 7 above)
Comment (krysal) [reply test/unit/specs/components/ImageDetails/v-copy-license.spec.js]: Heck, I was missing why they weren't working with the composition API but it turned out to be something simple in the end. Done!

**Comment by krysal** [comment]
Right, I fixed the `margin-top` of the **title-button** section, the padding of the **Copy text** button and tabs of the attribution box, plus worked on @sarayourfriend's comments. Also, I kept the old sketchfab viewer.

@panchovm I created an [issue](https://github.com/WordPress/openverse-frontend/issues/764) to update the styles for the License description items alone, as this PR may already fix some issues for the audio details page should be merged soon.

File: src/components/VImageDetails/VRelatedImages.vue (hunk 3 above)
Comment (sarayourfriend) [reply src/components/VImageDetails/VRelatedImages.vue:22]: Oops, for some reason I thought the new masonry layout was already merged, but I was thinking of the square layout 🤦‍♀️ 

Sorry for the distraction!

**Review by sarayourfriend — APPROVED** [review]
LGTM! I'm so excited for this, the improvement to the design and the code is so awesome! Kudos to both @krysal and @panchovm for this significant improvement 🎉

**Comment by panchovm** [comment]
The changes look awesome. Excellent work ✨ 

I was navigating the box attribution area and noticed the following.



My error not adding the tab component and its different states. The focus style looks good in **Plain text** tab but it seems better to set an internal border of `1.5px` to solve the overlapping effect. Then, we should skip the focus on the box area where attribution info and copy button are as it does nothing.

File: src/pages/image/_id.vue [hunk 8]
```diff
@@ -1,48 +1,143 @@
...
+      {{ $t('image-details.back') }}
+    </a>
+
+    <figure class="w-full mb-4 pt-4 md:pt-8 px-6 bg-dark-charcoal-06">
+      <img
+        v-if="!sketchFabUid"
+        id="main-image"
+        :src="image.url"
+        :alt="image.title"
+        class="h-full max-h-[500px] sm:mx-auto rounded-t-sm"
```
Comment (zackkrida) [inline src/pages/image/_id.vue:19]:
```suggestion
        class="h-full max-h-[500px] mx-auto rounded-t-sm"
```

This prefix breaks the center-alignment of the image on very small mobile screens.

**Review by zackkrida — APPROVED** [review]
This looks beautiful! 🥳 So much hard work here. I made one small suggestion to improve the appearance on very small displays that I'd love to see included, but other than that I'd like to get this merged right away.

I'm going to create a few new issues based on some things I've noticed here, unrelated to your work.

**Review by panchovm — COMMENTED** [review]
Review the focus style of attribution box. If the change belongs to a component improvement, then ignore the suggestion and merge it.

**Comment by krysal** [comment]
@panchovm I made the border internal with `box-shadow` and also @sarayourfriend created #749 to address more a11y issues with the tabs.

@zackkrida suggestions applied. Merging... 🚀

**Pull request merged**